caddyhttp/httpserver/server.go

```diff
@@ -380,6 +380,10 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	status, _ := s.serveHTTP(w, r)
 
+	if status == 204 {
+		w.WriteHeader(http.StatusNoContent)
+	}
```

Comment:
Hmm, probably we should instead write whatever `status` is returned instead of coding a special case.